joplin/pages/topic_collection_page/models.py

```diff
@@ -56,11 +55,13 @@ class JanisBasePageWithTopicCollections(JanisBasePage):
     def janis_urls(self):
         # Add the urls for each topic collection, these pages only
         # should publish at /theme_slug/topic_collection_slug/page_slug
-        topic_collections = self.topic_collections.all()
+        urls = []
 
-        return ['{topic_collection_url}{page_slug}/'.format(
-            topic_collection_url=base_page_topic_collection.topic_collection.janis_url(), page_slug=self.slug)
-            for base_page_topic_collection in topic_collections]
+        for base_page_topic_collection in self.topic_collections.all():
+            for topic_collection_url in base_page_topic_collection.topic_collection.janis_urls():
```

Comment:
am I reading this correctly, can a topic_collection have more than one url? Can they be under multiple themes?

Reply:
They cannot be under multiple themes. And a topic_collection cannot have more than one url. For now. I changed the code to iterate over the entire list (of 1) instead of just picking the first url. Functionally, the end result is the same, and our unit tests confirm that. But it leaves our janis_urls open to more flexibility in the future, should be require it.